index.js

```diff
@@ -55,22 +74,80 @@ async function init() {
 
 	console.log('Source:', {repo, ref, dir});
 
+	if (!navigator.onLine) {
+		document.querySelector('#error-network').style.display = 'block';
+		throw new Error('User agent is offline');
+	}
+
 	updateStatus('Retrieving directory info…');
 
+	await ensureRepoIsAccessible(repo);
```

Comment:
I'd rather see this request being made in parallel with the `trees` request. You can then use `Promise.all` to catch all the errors in one try/catch

Reply:
But really, you can probably skip this function altogether. 404 folder/repos will be caught by the `trees` API call and private repos support hopefully will come soon, right? :D

Reply:
Yeah, this originally was just a check for whether the repo is public. The reason I arranged it sequentially is to spare the comparatively large trees request if it's not needed. That's probably not a concern anyway if someone wants to download a ton of files. 🙈 

Then I wrapped the trees call in a `try/catch` to catch non-existing repos and finally merged those two approaches together and that's the result. 🤷‍♂️

But parallelizing it sounds reasonable, will do. Can always remove the check when support for private repos kicks in.

Reply:
Actually it's alright, it's not a big deal here. It'd just make things harder to read.